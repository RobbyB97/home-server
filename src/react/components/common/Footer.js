import React from 'react'


export class Footer extends React.Component {
    
    render() {
        return (
            <div id="footer">
                <div class="footer__logos">

                </div>
                <div class="footer__copyright">
                    
                </div>
            </div>
        )
    }
}


export default Footer