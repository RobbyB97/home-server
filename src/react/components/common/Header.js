import React from 'react'
import {NavLink} from 'react-router-dom'
import {connect} from 'react-redux'

import {toggleNav, toggleNavOff} from '../../actions/navigation'
import {setTitle} from '../../actions/title'
import {homeOn} from '../../actions/home'


const Header = ({nav, dispatch}) => (
	<header id="header">
		<div className="header__logo">
	  		<NavLink to="/react/" onClick={() => {
				dispatch(setTitle(''))
				dispatch(toggleNavOff())
				dispatch(homeOn())
	  		}}>
			<img src='/assets/images/favicon.png' alt='Home'/> 
	  		</NavLink>
		</div>

		<button className="header__nav-button" onClick={() => {
	  		dispatch(toggleNav())
		}}>
	  		{!nav ? '☰':'✕'}
		</button>    
  	</header>
) 

const mapStateToProps = (state) => {
	return {
		nav: state.navigation.nav
  	}
}

export default connect(mapStateToProps)(Header)