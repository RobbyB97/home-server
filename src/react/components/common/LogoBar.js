import React from 'react'

const LogoBar = () => (
    <div id="logo-bar">
        <ul>
            <li>
                <a href="https://www.github.com/robbyb97" target="_blank">
                    <img src="/dist/images/icons/github.svg" />
                </a>
            </li>
            
            <li>
                <a href="https://www.linkedin.com/in/robbyb97" target="_blank">
                    <img src="/dist/images/icons/linkedin.png" />
                </a>
            </li>
        </ul>
    </div>
)

export default LogoBar