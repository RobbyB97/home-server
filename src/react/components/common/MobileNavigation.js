import React from 'react'
import {NavLink} from 'react-router-dom'
import { connect } from 'react-redux'

import {toggleNav, toggleNavOff} from '../../actions/navigation'
import {setTitle} from '../../actions/title'
import {homeOn, homeOff} from '../../actions/home'


const MobileNavigation = ({nav, dispatch, home}) => (
    <div id="mobile-navigation" className={nav ? 'active-nav':''}>
        <div className="mobile-navigation__background">
            <ul className="mobile-navigation__intra-nav">
                <li>
                    <NavLink to="/" onClick={() => {
                        dispatch(toggleNav())
                        dispatch(setTitle(''))
                        dispatch(homeOn())
                    }}>Home</NavLink>
                </li>

                <li>
                    <NavLink to="/skills/" onClick={() => {
                        dispatch(toggleNav())
                        dispatch(setTitle('Skills'))
                        dispatch(homeOff())
                    }}>Skills</NavLink>
                </li>

                <li>
                    <NavLink to="/experience/" onClick={() => {
                        dispatch(toggleNav())
                        dispatch(setTitle('Experience'))
                        dispatch(homeOff())
                    }}>Experience</NavLink>
                </li>

                <li>
                    <NavLink to="/projects/" onClick={() => {
                        dispatch(toggleNav())
                        dispatch(setTitle('Projects'))
                        dispatch(homeOff())
                    }}>Projects</NavLink>
                </li>

                <li>
                    <NavLink to="/contact/" onClick={() => {
                        dispatch(toggleNav())
                        dispatch(setTitle('Contact'))
                        dispatch(homeOff())
                    }}>Contact</NavLink>
                </li>
            </ul>
        </div>
    </div>
)

const mapStateToProps = (state) => {
    return {
        nav: state.navigation.nav,
        home: state.home.home
    }
}

export default connect(mapStateToProps)(MobileNavigation)