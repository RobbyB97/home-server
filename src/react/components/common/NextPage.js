import React from 'react'
import {NavLink} from 'react-router-dom'


export class NextPage extends React.Component {
    

    render() {
        return (
            <div className={this.props.float ? "nextPage float" : "nextPage"}>
                <NavLink 
                    to={this.props.route ? this.props.route : '/'}
                >
                    <p className="nextPage__text">
                        {this.props.text && this.props.text} â†’
                    </p>
                </NavLink>
            </div>
        )
    }
}


export default NextPage