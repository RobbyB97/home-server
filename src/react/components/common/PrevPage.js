import React from 'react'
import {NavLink} from 'react-router-dom'


export class PrevPage extends React.Component {
    

    render() {
        return (
            <div className="prevpage">
                {this.props.text &&
                    <p className="prevpage__text">
                        {this.props.text}
                    </p>
                }
                <NavLink 
                    to={this.props.route ? this.props.route : '/'}
                >
                    <div className="prevpage__arrow">
                        <img src="/dist/images/icons/down_arrow_white.png"/>
                    </div>
                </NavLink>
            </div>
        )
    }
}


export default PrevPage