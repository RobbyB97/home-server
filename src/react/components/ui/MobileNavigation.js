import React from 'react'
import {connect} from 'react-redux'

import {navToggle} from '../../actions/ui'


export class MobileNavigation extends React.Component {
    constructor(props) {
        super(props);

        this.navToggle = this.navToggle.bind(this);
        this.body = document.getElementsByTagName("BODY")[0];
        this.html = document.getElementsByTagName("HTML")[0];
    }

    navToggle = () => {
        // Reset scroll position
        if (this.props.nav) {   // Turning nav off
            this.body.style.overflow = "unset";
            this.html.style.overflow = "unset";
            this.body.style.overflowX = "hidden";

        } else {    // Turning nav on
            this.body.style.overflow = "hidden";
            this.html.style.overflow = "hidden";
        }

        this.props.navToggle();
    }

    render() {
        return (
            <section id="MobileNavigation">
                <div className="mobileNavigation__button"
                    data-nav={this.props.nav}
                >
                    <div onClick={this.navToggle}>
					    {this.props.nav ? '✕':'☰'}
                    </div>
				</div>

                <nav className="mobileNavigation__menu"
                    data-nav={this.props.nav}
                >
                    <ul className="mobileNavigation__links">
                        <li>
                            <a href="#About" onClick={this.navToggle}>About</a>
                        </li>
                        
                        <li>
                            <a href="#Skills" onClick={this.navToggle}>Skills</a>
                        </li>

                        <li>
                            <a href="#Projects" onClick={this.navToggle}>Projects</a>
                        </li>

                        <li>
                            <a href="#Experience" onClick={this.navToggle}>Experience</a>
                        </li>
                    </ul>
                </nav>
            </section>
        );
    }
}


const mapStateToProps = (state) => {
    return {
        nav: state.ui.nav,
        white: state.ui.white
    }
};

const mapDispatchToProps = (dispatch) => ({
    navToggle: () => {
        dispatch(navToggle());
    }
});


export default connect(mapStateToProps, mapDispatchToProps)(MobileNavigation);