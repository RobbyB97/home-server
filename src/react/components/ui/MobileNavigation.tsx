export {}

/* Libraries */
import * as React from 'react';
import {connect} from 'react-redux';

import {navToggle} from '../../actions/ui';


/* Component */
interface Props {
    nav: boolean,
    navToggle: typeof navToggle
}
interface State {
    nav: boolean,
    navToggle: typeof navToggle
}

export class MobileNavigation extends React.Component<Props, State> {
    constructor(props: any) {
        super(props);
        this.state = {
            ...this.state
        }
    }

    navToggle = () => {
        this.state.navToggle();
    };

    public render() {
        const {nav} = this.state;
        const {navToggle} = this.state;

        return (
            <div id="MobileNavigation">
                <div className={nav ?
                    "mobileNavigation__button nav":
                    "mobileNavigation__button"}>
                    <div onClick={navToggle}>
					    {nav ? '✕':'☰'}
                    </div>
				</div>

                <div className={nav ? 
                    "mobileNavigation__menu nav":
                    "mobileNavigation__menu"}>
                    <ul className="mobileNavigation__links">
                        <li>
                            <a href="#about" onClick={navToggle}>about</a>
                        </li>
                        <li>
                            <a href="#services" onClick={navToggle}>services</a>
                        </li>
                        <li>
                            <a href="#skills" onClick={navToggle}>skills</a>
                        </li>
                        <li>
                            <a href="#experience"onClick={navToggle}>experience</a>
                        </li>
                        <li>
                            <a href="#projects" onClick={navToggle}>projects</a>
                        </li>
                    </ul>
                </div>
            </div>
        );
    }
}

/* Connect to store */
const mapStateToProps = (state) => ({
    nav: state.ui.nav,
});

const mapDispatchToProps = (dispatch) => ({
    navToggle: () => {
        dispatch(navToggle())
    }
});


/* Export */
export default connect(mapStateToProps, mapDispatchToProps)(MobileNavigation);