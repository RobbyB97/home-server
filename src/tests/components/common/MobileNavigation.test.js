import React from 'react'
import {shallow} from 'enzyme'

import {MobileNavigation} from '../../../react/components/common/MobileNavigation'


let wrapper, navHome, navNotHome

beforeEach(() => {
    navHome = jest.fn()
    navNotHome = jest.fn()
    wrapper = shallow(
        <MobileNavigation
            navHome={navHome}
            navNotHome={navNotHome}
        />
    )
})


test('Render MobileNavigation', () => {
    expect(wrapper).toMatchSnapshot()
})


test('Call navHome', () => {
    wrapper.find('NavLink').at(0).simulate('click')

    expect(navHome).toHaveBeenCalled()
})


test('Call navNotHome', () => {
    wrapper.find('NavLink').at(1).simulate('click')

    expect(navNotHome).toHaveBeenCalled()
})