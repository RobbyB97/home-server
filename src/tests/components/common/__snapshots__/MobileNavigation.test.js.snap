// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render MobileNavigation 1`] = `
<div
  className=""
  id="mobile-navigation"
>
  <div
    className="mobile-navigation__background"
  >
    <ul
      className="mobile-navigation__intra-nav"
    >
      <li>
        <NavLink
          activeClassName="active"
          ariaCurrent="true"
          onClick={[Function]}
          to="/"
        >
          Home
        </NavLink>
      </li>
      <li>
        <NavLink
          activeClassName="active"
          ariaCurrent="true"
          onClick={[Function]}
          to="/skills/"
        >
          Skills
        </NavLink>
      </li>
      <li>
        <NavLink
          activeClassName="active"
          ariaCurrent="true"
          onClick={[Function]}
          to="/experience/"
        >
          Experience
        </NavLink>
      </li>
      <li>
        <NavLink
          activeClassName="active"
          ariaCurrent="true"
          onClick={[Function]}
          to="/projects/"
        >
          Projects
        </NavLink>
      </li>
      <li>
        <NavLink
          activeClassName="active"
          ariaCurrent="true"
          onClick={[Function]}
          to="/resume"
        >
          Resume
        </NavLink>
      </li>
      <li>
        <a
          href="mailto:bergersr@my.easternct.edu"
        >
          Contact
        </a>
      </li>
    </ul>
  </div>
</div>
`;
